browser/net/brave_torrent_redirect_network_delegate_helper.cc

```diff
@@ -44,34 +48,40 @@ bool IsMainFrameResource(std::shared_ptr<brave::BraveRequestInfo> ctx) {
 
 namespace webtorrent {
 
+bool ShouldRedirectRequest(
+    const net::HttpResponseHeaders* original_response_headers,
+    std::shared_ptr<brave::BraveRequestInfo> ctx) {
+  return !(!original_response_headers || !IsMainFrameResource(ctx) ||
+           ctx->is_webtorrent_disabled ||
+           // download .torrent, do not redirect
+           (IsWebtorrentInitiated(ctx) && !IsViewerURL(ctx->request_url)) ||
+           !IsTorrentFile(ctx->request_url, original_response_headers));
+}
+
 int OnHeadersReceived_TorrentRedirectWork(
     const net::HttpResponseHeaders* original_response_headers,
     scoped_refptr<net::HttpResponseHeaders>* override_response_headers,
     GURL* allowed_unsafe_redirect_url,
     const brave::ResponseCallback& next_callback,
     std::shared_ptr<brave::BraveRequestInfo> ctx) {
-
-  if (!original_response_headers ||
-      !IsMainFrameResource(ctx) ||
-      ctx->is_webtorrent_disabled ||
-      // download .torrent, do not redirect
-      (IsWebtorrentInitiated(ctx) && !IsViewerURL(ctx->request_url)) ||
-      !IsTorrentFile(ctx->request_url, original_response_headers)) {
+  if (!ShouldRedirectRequest(original_response_headers, ctx)) {
     return net::OK;
   }
 
-  *override_response_headers =
-    new net::HttpResponseHeaders(original_response_headers->raw_headers());
-  (*override_response_headers)
-      ->ReplaceStatusLine("HTTP/1.1 307 Temporary Redirect");
-  (*override_response_headers)->RemoveHeader("Location");
-  GURL url(
-      base::StrCat({extensions::kExtensionScheme, "://",
-      brave_webtorrent_extension_id,
-      "/extension/brave_webtorrent2.html?",
-      ctx->request_url.spec()}));
-  (*override_response_headers)->AddHeader("Location", url.spec());
-  *allowed_unsafe_redirect_url = url;
+  auto* contents =
+      content::WebContents::FromFrameTreeNodeId(ctx->frame_tree_node_id);
+  GURL url(base::StrCat(
+      {extensions::kExtensionScheme, "://", brave_webtorrent_extension_id,
+       "/extension/brave_webtorrent2.html?", ctx->request_url.spec()}));
+  content::NavigationController::LoadURLParams params(url);
+  params.is_renderer_initiated = true;
```

Comment:
why this is needed? can you write a comment please

Reply:
`is_renderer_initiated` doesn't seem to be, but I was cargo culting the API call from inside the Chrome Extensions API.

I've added a comment explaining why we need to do this.

Reply:
but i guess we need either is_renderer_initiated and friends (initiator, etc below) or none of this?

Reply:
in case it has been cargoculted from here https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/extensions/api/tabs/tabs_api.cc;l=1671;drc=abbfc06e9b070c9ea64af2b233624646cef00252;bpv=1;bpt=1

Reply:
So the only required field here is `transition_type` (without it `magnet` urls break). Do you have a preference about removing the other stuff/leaving as is?